src/contact-details/contact-details.component.tsx

```diff
@@ -0,0 +1,148 @@
+import React from "react";
+import InlineLoading from "carbon-components-react/es/components/InlineLoading";
+import { createErrorHandler } from "@openmrs/esm-framework";
+import {
+  fetchPatientRelationships,
+  Relationship
+} from "./relationships.resource";
+import styles from "./contact-details.scss";
+import { useTranslation } from "react-i18next";
+
+const Address: React.FC<{ address: fhir.Address }> = ({ address }) => {
+  const { city, country, postalCode, state } = address;
+  const { t } = useTranslation();
+
+  return (
+    <div className={styles.col}>
+      <p className={styles.heading}>{t("address", "Address")}</p>
+      <ul>
+        <li>{postalCode}</li>
+        <li>{city}</li>
+        <li>{state}</li>
+        <li>{country}</li>
+      </ul>
+    </div>
+  );
+};
+
+const Contact: React.FC<{ telecom: Array<fhir.ContactPoint> }> = ({ telecom }) => {
+  const value = telecom ? telecom[0].value : "-";
+  const { t } = useTranslation();
+  return (
+    <div className={styles.col}>
+      <p className={styles.heading}>{t("contactDetails", "Contact Details")}</p>
+      <ul>
+        <li>{value}</li>
+      </ul>
+    </div>
+  );
+};
+
+const Relationships: React.FC<{ patientId: string }> = ({ patientId }) => {
+  const [relationships, setRelationships] = React.useState<
+    Array<ExtractedRelationship>
+  >(null);
+
+  React.useEffect(() => {
+    function extractRelationshipData(
+      relationships: Array<Relationship>
+    ): Array<ExtractedRelationship> {
+      const relationshipsData = [];
+      for (const r of relationships) {
+        if (patientId === r.personA.uuid) {
+          relationshipsData.push({
+            uuid: r.uuid,
+            display: r.personB.person.display,
+            relativeAge: r.personB.person.age,
+            relativeUuid: r.personB.uuid,
+            relationshipType: r.relationshipType.bIsToA
+          });
+        } else {
+          relationshipsData.push({
+            uuid: r.uuid,
+            display: r.personA.person.display,
+            relativeAge: r.personA.person.age,
+            relativeUuid: r.personA.uuid,
+            relationshipType: r.relationshipType.aIsToB
+          });
+        }
+      }
+      return relationshipsData;
+    }
+
+    fetchPatientRelationships(patientId)
+      .then(({ data: { results } }) => {
+        if (results.length) {
+          const relationships = extractRelationshipData(results);
+          setRelationships(relationships);
+        }
+      })
+      .catch(createErrorHandler());
+  }, [patientId]);
+
+  const RenderRelationships: React.FC = () => {
```

Comment:
Why is this component captured? Please don't re-create components in renderings, as this will *always* re-render (no matter if props changed).